### ISSUE TITLE
(documentation?) index.delete requires unique ids

### ISSUE THREAD

**Issue by thehesiod** [Toblerity/rtree#76]
per http://toblerity.org/rtree/class.html#rtree.index.Index.delete it states for the id parameter to delete: ```long integer A long integer that is the identifier for this index entry. IDs need not be unique to be inserted into the index, and it is up to the user to ensure they are unique if this is a requirement.```

However I tried this with with an Index which had a set of points which all had the same ids and it never removed it items.  As soon as I gave each point a unique id the points got removed.

**Comment by sr-murthy** [comment]
@thehesiod I assume this remains an open issue - if so can I work on it?

**Comment by hobu** [comment]
This is still an open issue, but I think it is a documentation one. libspatialindex indexes by bounding boxes + id, and to delete a unique record, you need to provide both.

**Comment by sr-murthy** [comment]
Yes, this makes sense because insertion of entries requires a (not necessarily unique) ID and a bounding box. It does seem strange that insertion of duplicate entries (entries with identical IDs and bounding boxes) does not throw an error - should this in fact be addressed first before the delete issue?

    In [1]: from rtree.index import Index

    In [2]: idx = Index()

    In [3]: idx.insert(1, (0.0, 0.0, 0.0, 0.0), obj=1)                                                                               

    In [4]: idx.insert(1, (0.0, 0.0, 0.0, 0.0), obj=1)                                                                               

    In [5]: idx.leaves()                                                                                                             
    Out[5]: [(0, [1, 1], [0.0, 0.0, 0.0, 0.0])] 

Why is this allowed in the first place?

**Comment by hobu** [comment]
libspatialindex allows it. It is up to user applications to prevent it from happening.

**Comment by thehesiod** [comment]
If it's a doc problem that is fine, let's get it fixed :)

**Comment by sr-murthy** [comment]
OK, I'm happy to work on it.

**Comment by sr-murthy** [comment]
I've created a PR here

https://github.com/Toblerity/rtree/pull/133

that clarifies the docstring for `index.Index.delete`. I hope it addresses the documentation side of this issue. Clearly, as the index allows insertion of multiple items/entries with identical IDs and coordinates, it is up to the user to enforce uniqueness at the application level.

**Comment by sr-murthy** [comment]
@hobu @thehesiod What should be done about the error/exception issue?

**Comment by thehesiod** [comment]
That's a question for hobu

**Comment by hobu** [comment]
I think documentation is sufficient. I don't think Rtree should throw an exception in this instance. It is possible users have a reason to insert duplicates, and they should protect against that themselves if they need to do so.
